script.motr/resources/lib/kodiutils.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import sys
+import logging
+
+# read settings
+ADDON = xbmcaddon.Addon()
+__lang__ = ADDON.getLocalizedString
+
+def notification(header, message, time=5000, icon=ADDON.getAddonInfo('icon'), sound=True):
+    xbmcgui.Dialog().notification(header, message, icon, time, sound)
+
+def errornotification(message, time=5000, icon=ADDON.getAddonInfo('icon'), sound=True):
+    xbmcgui.Dialog().notification(__lang__(30100), message, icon, time, sound)
+    
+def dialogok(header, line1, line2='', line3=''):
+    xbmcgui.Dialog().ok(header, line1, line2, line3)
+
+def dialogyesno(header, line1, line2='', line3=''):
+    return xbmcgui.Dialog().yesno(header, line1, line2, line3) #True on yes
+    
+def dialogokerror(line1, line2='', line3=''):
+    xbmcgui.Dialog().ok(__lang__(30101), line1 + "[CR]" + line2 + "[CR]" + line3)
+
+def dialogstreamordownload(header):
+    return xbmcgui.Dialog().select(header, [__lang__(30102), __lang__(30114), __lang__(30103), __lang__(30007)])
+
+def dialogunpackordownload(header):
+    return xbmcgui.Dialog().select(header, [__lang__(30104), __lang__(30007)])
+    
+def dialogqueuerunning():
+    return xbmcgui.Dialog().select(__lang__(30011), [__lang__(30105), __lang__(30106), __lang__(30107)])
+
+def dialogqueuenotrunning():
+    return xbmcgui.Dialog().select(__lang__(30011), [__lang__(30108), __lang__(30109), __lang__(30110), __lang__(30111), __lang__(30112), __lang__(30113)])
+    
+def dialogtext(header, text):
+    xbmcgui.Dialog().textviewer(header, text)
+    
+def show_settings():
+    ADDON.openSettings()
+
+def get_setting(setting):
+    return ADDON.getSetting(setting).strip()
+
+def set_setting(setting, value):
+    ADDON.setSetting(setting, str(value))
+
+def get_setting_as_bool(setting):
```

Comment:
all these methods don't make any sense anymore. We have getSettingBool, getSettingInt, etc on the python api:
https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html#gaf0205dd5b96354f30a39fbf35599e681

Reply:
Depending on how you see it. By doing so you have single point for set / get settings.
Makes it much simpler if you want to add some kind of logic later on. Like the get_setting always strip the value it gets.

In my taste for example get_string(id) function is better than each time calling xbmcaddon.Addon().getLocalizedString(id)
And it basically the same thing been called, since ADDON = xbmcaddon.Addon(), then get_string(id) calls ADDON.getLocalizedString(string_id).

I guess this isn't directly wrong so I would like to have it like that, if it is OK for you?

Reply:
This was just a recommendation, not mandatory to change :)